### ISSUE TITLE
[BUG] Codemode breaks react-router

### ISSUE THREAD

**Issue by Mikilll94** [refinedev/refine#3926]
### Describe the bug

After running codemode

```
npx @refinedev/codemod@latest refine3-to-refine4
```

I got the following error:
```
Module not found: Error: Can't resolve 'react-router-dom' in '/Users/user/my/europig-refine/src/users'
```

This is because of this change done by codemode:

![Screenshot 2023-03-20 at 12 12 24](https://user-images.githubusercontent.com/17653648/226323273-2e64a19a-504a-432a-b3f3-0d18c95fc81f.png)

### Steps To Reproduce

1. Add this line somewhere in your codebase:
```
import { useNavigate } from "@pankod/refine-react-router-v6";
```
2. Run this codemode
```
@refinedev/codemod@latest refine3-to-refine4
```
3. Run the project. Error will be thrown.

### Expected behavior

Codemode should not break react-router

### Desktop

- macOS 11.6.1

**Comment by BatuhanW** [comment]
Thanks for the report @Mikilll94, we are on it. For now, you can manually add `react-router-dom` to your package.json to circumvent the issue.

**Comment by BatuhanW** [comment]
Hello again @Mikilll94,

Can you confirm that you had `"@pankod/refine-react-router-v6";` dependency in your package.json before running codemod? I couldn't reproduce the issue.

**Comment by Mikilll94** [comment]
Yes, I had `@pankod/refine-react-router-v6` in my `package.json`.

This is what codemod did in my `package.json`:

![Screenshot 2023-03-20 at 13 04 50](https://user-images.githubusercontent.com/17653648/226334094-36bec246-feac-42fe-aa61-2ff7f28c7e4a.png)

**Comment by BatuhanW** [comment]
Hello @Mikilll94,

We have this line to check if project has `@pankod/react-router-v6` in the package.json and adds `react-router-dom` dependency to new package.json here https://github.com/refinedev/refine/blob/c682cfe5cf8f056d8773e2f23481d0d6d5488c4b/packages/codemod/src/transformations/v4/separate-imports-react-router-v6.ts#L17

I tried to run codemod on v3 version of our `base-mui` example and it added `react-router-dom` to package.json successfully, it's quite an interesting bug.

**Comment by BatuhanW** [comment]
This is the package.json diff after running codemod couple of times. Unfortunately, can't reproduce the issue. 

![Screenshot 2023-03-20 at 15 34 17](https://user-images.githubusercontent.com/16444991/226340134-fe295056-786e-41dd-b65b-b7d70e0f735a.png)

Could you provide us a reproducible example so we can debug? 

Closing the issue for now, feel free to re-open it if needed.
